### ISSUE TITLE
DeeplabV3+ Training on custom dataset

### ISSUE THREAD

**Issue by parth29-vc** [tensorflow/tensorflow#53804]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- Platform and Distribution: Colab
- TensorFlow version (use command below): 2.7.0
- Python version: 3.8
- CUDA/cuDNN version: 11.2
- GPU model and memory: Tesla P100, 16 GB memory

**Current situation**
 Worked on Semantics segmentation using TensorFlow's DeepLabv3+ with the Resnet versions backbone architecture. We've trained it for our custom dataset which includes satellite images of Golf Courses extracted from the Google Map API. Our Dataset includes 109 samples for training and we have trained it and got a decent amount of accuracy (~93%) but the results are not as expected.

**Current behavior**
Currently, we've referenced the test set which includes similar types of images but the results are not as good as expected. For the sake of curiosity, we have checked by inferencing on Train images also but that also not promising.

**Expected behavior**
Expected results with this accuracy should be descent as trained on the other custom dataset.

**Comment by sushreebarsa** [comment]
@parth29-vc 
In order to expedite the trouble-shooting process, please provide a code snippet to reproduce the issue reported here. Thanks!

**Comment by parth29-vc** [comment]
@sushreebarsa 
The gist for the code snippet is available here. [Gist](https://gist.github.com/parth29-vc/29029bd2989bfc994fedfb976af65b16)

**Comment by sushreebarsa** [comment]
@parth29-vc 
Please post this issue on [keras-team/keras repo.](https://github.com/keras-team/keras/issues)
To know more see;
[https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999](https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999)
Thanks!

**Comment by parth29-vc** [comment]
Thanks, added to Keras

**Comment by sushreebarsa** [comment]
@parth29-vc Thank you for the update!
Closing this issue as we will track the other ticket in keras-team/keras repo.
